packages/server/lib/cloud/api/putProtocolArtifact.ts

```diff
@@ -0,0 +1,28 @@
+import fsAsync from 'fs/promises'
+import fs from 'fs'
+import Debug from 'debug'
+import { uploadStream, geometricRetry } from '../upload/uploadStream'
+import { StreamActivityMonitor } from '../upload/StreamActivityMonitor'
+
+const debug = Debug('cypress:server:cloud:api:protocol-artifact')
+
+export const putProtocolArtifact = async (artifactPath: string, maxFileSize: number, destinationUrl: string) => {
+  debug(`Atttempting to upload Test Replay archive from ${artifactPath} to ${destinationUrl})`)
+  const { size } = await fsAsync.stat(artifactPath)
+
+  if (size > maxFileSize) {
+    throw new Error(`Spec recording too large: artifact is ${size} bytes, limit is ${maxFileSize} bytes`)
+  }
+
+  const activityMonitor = new StreamActivityMonitor(5000, 5000)
```

Comment:
any qualms with making these arguments an object so they read a bit clearer to their intent vs magic numbers?
```suggestion
  const activityMonitor = new StreamActivityMonitor({
       maxStartDwellTime: 5000,
       maxActivityDwellTime: 5000
  })
```

Reply:
Nope, I like it!

Reply:
Actually, it's a little unwieldy in practice - what if the magic numbers were consts instead? When referencing, the intent is fairly clear via IDE hints, but I understand that reading is another thing entirely

Reply:
That works for me!